### PR TITLE
Rename WeightNorm -> WeightNormalization

### PR DESCRIPTION
**Pull request by lgeiger**
This renames the `WeightNorm` wrapper to `WeightNormalization` in order to be consistent with  [`tf.keras.layers.BatchNormalization`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/normalization.py#L774).

See https://github.com/tensorflow/addons/pull/14#discussion_r248224896

### CODE REVIEW

**Comment by seanpmorgan** [comment]
Thanks. I have another PR in the works to optimize this for TF2, but no need to wait to rename
